clang/lib/Driver/Driver.cpp

```diff
@@ -1189,18 +1190,19 @@ static bool addSYCLDefaultTriple(Compilation &C,
   llvm::Triple DefaultTriple =
       C.getDriver().getSYCLDeviceTriple(getDefaultSYCLArch(C));
   for (const auto &SYCLTriple : SYCLTriples) {
+    // Check current set of triples to see if the default has already been set.
     if (SYCLTriple == DefaultTriple)
       return false;
-    // If we encounter a known non-spir* target, do not add the default triple.
-    if (SYCLTriple.isNVPTX() || SYCLTriple.isAMDGCN())
-      return false;
-  }
-  // Check current set of triples to see if the default has already been set.
-  for (const auto &SYCLTriple : SYCLTriples) {
+    // TODO: Do we need this check? Don't we already detect the default triple
+    // in the previous condition?
```

Comment:
Since we allow for spir64 or spirv64, this checks for the latter.  Perhaps we can just use this check and not have the one above?

Reply:
Good point. I forgot that we have two spellings to produce SPIR-V binaries. I'll try to avoid unnecessary checks.